### PR TITLE
Migrate splitRe to native function

### PR DESCRIPTION
**Pull request by nucliweb**
- Migrate [splitRe](https://github.com/substack/node-resolve/blob/master/lib/node-modules-paths.js#L20) to native function [path.sep](https://nodejs.org/api/path.html#path_path_sep)
- Remove unused [variable](https://github.com/substack/node-resolve/blob/master/lib/async.js#L6)
- Update to two last versions of Node in .travis.yml

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 language: node_js
 node_js:
-  - 0.6
-  - 0.8
+  - "5.4.0"
```
Comment (ljharb) [inline .travis.yml]: please revert this; it's both already done in master and not relevant to this PR

File: lib/async.js [hunk 2]
```diff
@@ -55,23 +54,23 @@ module.exports = function resolve (x, opts, cb) {
             else cb(new Error("Cannot find module '" + x + "' from '" + y + "'"))
         })
     }
-    
+
```
Comment (ljharb) [inline lib/async.js]: please revert all the trailing whitespace removal on lines that you're not already changing

**Comment by ljharb** [comment]
Please rebase so as to avoid merge commits - if you check the "allow edits" box, I can take care of it for you.

**Comment by nucliweb** [comment]
Done!

**Comment by ljharb** [comment]
@nucliweb there's still a merge commit - you'll need to rebase on top of latest master on the command line to remove it.

**Comment by nucliweb** [comment]
I checked "Allow edits from maintainers"

**Comment by ljharb** [comment]
@nucliweb thanks, it's all rebased - but there's a legitimate test failure on node 0.6 now.

**Comment by nucliweb** [comment]
🤔 We want support to version 0.6?

**Comment by ljharb** [comment]
@nucliweb yes, this package supports 0.6 and will continue to do so until there's a good reason to actively drop support for it - and a minor refactor isn't one.

**Comment by nucliweb** [comment]
@ljharb So we can not merge this PR 😓

**Comment by ljharb** [comment]
@nucliweb not without a fix for 0.6 - overall though I do prefer the use of `path.sep`, so I'd love to see this be able to get in.

**Comment by ljharb** [comment]
This will need to be rebased after #113 goes in.

**Comment by ljharb** [comment]
Thanks for your contribution! However, now that we're using `path.parse` in #113, this PR no longer applies.

**Pull request closed without merging**